Libraries/Components/ActivityIndicator/ActivityIndicator.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ActivityIndicator
+ * @flow
+ */
+'use strict';
+
+const ColorPropType = require('ColorPropType');
+const NativeMethodsMixin = require('NativeMethodsMixin');
+const Platform = require('Platform');
+const PropTypes = require('ReactPropTypes');
+const React = require('React');
+const StyleSheet = require('StyleSheet');
+const View = require('View');
+
+const requireNativeComponent = require('requireNativeComponent');
+
+const GRAY = '#999999';
+
+/**
+ * Displays a circular loading indicator.
+ */
+const ActivityIndicator = React.createClass({
+  mixins: [NativeMethodsMixin],
+
+  propTypes: {
+    ...View.propTypes,
+    /**
+     * Whether to show the indicator (true, the default) or hide it (false).
+     */
+    animating: PropTypes.bool,
+    /**
+     * The foreground color of the spinner (default is gray).
+     */
+    color: ColorPropType,
+    /**
+     * Size of the indicator. Small has a height of 20, large has a height of 36.
+     * Other sizes can be obtained using a scale transform.
+     */
+    size: PropTypes.oneOf([
```

Comment:
I wasn't sure exactly what sizes we want to support here. For iOS there are only 2 sizes possible, the only way to get more is to use a scale transform. For Android there were a few more size presets and it's possible to get any size by passing `width` and `height` to `RCTActivityIndicator`.

I used the same sizes for Android that iOS uses which mean that some of the size presets from ProgressBarAndroid are different / missing. The advantage of this is that the spinners will have the same size on both platform and it is still possible make the spinners any size using a scale transform.